pkg/transformers/neosync/neosync_email_transformer.go

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package neosync
+
+import (
+	"fmt"
+
+	neosynctransformers "github.com/nucleuscloud/neosync/worker/pkg/benthos/transformers"
+	"github.com/xataio/pgstream/pkg/transformers"
+)
+
+type EmailTransformer struct {
+	*transformer[string]
+}
+
+func NewEmailTransformer(params transformers.Parameters) (*EmailTransformer, error) {
+	preserveLength, err := findParameter[bool](params, "preserve_length")
+	if err != nil {
+		return nil, fmt.Errorf("neosync_email: preserve_length must be a boolean: %w", err)
+	}
+
+	preserveDomain, err := findParameter[bool](params, "preserve_domain")
+	if err != nil {
+		return nil, fmt.Errorf("neosync_email: preserve_domain must be a boolean: %w", err)
+	}
+
+	excludedDomains, err := findParameter[any](params, "excluded_domains")
```

Comment:
Might be good to add a comment here to say that neosync expects the `excluded_domains` as type `any`. Otherwise this should normally be a `[]string` and it can be misleading.